src/babel_utils.py

```diff
@@ -227,9 +227,12 @@ def write_compendium(synonym_list,ofname,node_type,labels={},extra_prefixes=[]):
                 nw['identifiers'] = [ {k[0]:v for k,v in nids.items()} for nids in node['identifiers']]
                 outf.write( nw )
                 synonyms = synonym_factory.get_synonyms(node)
-                if len(synonyms) > 0:
-                    for p,o in synonyms:
-                        sfile.write(f'{node["identifiers"][0]["identifier"]}\t{p}\t{o}\n')
+                synonyms.sort(key=lambda x:len(x))
+                document = {"curie": node["identifiers"][0]["identifier"],
+                            "preferred_name": node["identifiers"][0]["label"],
+                            "name": synonyms,
```

Comment:
If this is going to be a list of synonyms and/or labels, I think the property name should be `names` rather than `name`.